survey/docker.py

```diff
@@ -55,7 +58,10 @@ async def read_response():
             results = []
             for _ in purls:
                 line = await self.process.stdout.readline()
-                line = json.loads(line)
+                try:
+                    line = json.loads(line)
+                except Exception:
```

Comment:
```suggestion
                except json.JSONDecodeError:
```
This will help to be more specific, so as not to break on other exceptions...like keyboard interrupt. See [docs here](https://devdocs.io/python~3.10/library/json#json.loads).

Reply:
Keyboard interrupts are base exceptions, not exceptions.

Reply:
Ah...right. Hadn't reviewed [the list](https://docs.python.org/3/library/exceptions.html#exception-hierarchy) in a while...mostly just going off [the `ruff` finding](https://docs.astral.sh/ruff/rules/blind-except/).

Reply:
I think for languages with unchecked exception-based error handling, especially when everything is mutable, you should often catch broadly in cases like this where you're putting in a catch because you want to stop errors from a block of code.

The Python documentation is wrong. `json.loads` by default uses `float` to parse numbers with periods or scientific notation. `float` is documented to raise `OverflowError` and I don't see any handling to turn that into a `JSONDecodeError`, but my `float` is broken and returns infinity for large values instead of telling me there was an overflow so I can't demonstrate the problem with `json.loads`.

I've had this kind of problem with Python in the past. Sometimes when using Python networking APIs instead of the Python exception you'd expect you get an `OSError` instead of the correct Python exception type and you need to check the error number if you care about what happened.